Assets/Scripts/Switch.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Switch : MonoBehaviour
+{
+    [SerializeField]
+    [Tooltip("The color of the switch. 0 if yellow, 1 if blue, 2 if red, 3 if green.")]
+    [Range(0, 3)]
+    private int switchColor;
+
+    [SerializeField]
+    [Tooltip("The puzzle this switch is connected to.\nMust have a script which implements the IPuzzle interface.")]
+    private GameObject connectedPuzzle;
+
+    [SerializeField]
+    [Tooltip("The character this switch adds to the solution.")]
+    private char solutionEntry = '1';
+
+    /// <summary>
+    /// The animator component attached to the switch
+    /// </summary>
+    private Animator animator;
+
+    /// <summary>
+    /// Is the switch active or not?
+    /// </summary>
+    private bool isSwitchOn;
+
+    /// <summary>
+    /// The IPuzzle component of the connected puzzle
+    /// </summary>
+    private IPuzzle puzzle;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        animator = GetComponent<Animator>();
+        animator.SetInteger("Color", switchColor);
+        puzzle = connectedPuzzle.GetComponent<IPuzzle>();
+        puzzle.AppendSwitchList(this);
+    }
+
+    /// <summary>
+    /// Sends the switch's solution to the puzzle and turns it on
+    /// Called by OnUse from the Dialogue System Usable component
```

Comment:
When you are writing methods to handle events, create an "event handler" method named after the event: "OnUse()" and then inside of that method call whatever other methods you want to happen at that point (e.g. SetSwitch). This way SetSwitch can remain private, and I can understand the connection between a method and an event more easily.